latitudesh/terraform-provider-latitudesh
# fix: sshkeys and project id type and update docs

#### What does this PR do?
This PR aims to solve some know problems with our provider
- The server sshkeys attribute type
- The ServerProject.ID type

#### Description of Task to be completed?
Update sshkeys and ServerProject.ID type and some general updates to comply with our Go-SDK v0.2.6

#### How should this be manually tested?
- Installation
Before installing you will need to modify the `Makefile` present in the folder with the right information.
Normally you'll only need to modify two fields:
```
VERSION=0.0.1 # Make sure this doesn't interfere with other installations that you might have. I leave it as 0.0.1
OS_ARCH=linux_amd64 # This is your operating system architecture
```
*You can find the full list of suported `OS_ARCH` at `https://github.com/latitudesh/terraform-provider-latitudesh/releases` by expanding Assets.*

Them run `make install` to do a local installation of the provider.

- Setup
Create a new folder with a `main.tf` file:
```
# main.tf
terraform {
  required_providers {
    latitudesh = {
      source  = "latitude.sh/iac/latitudesh"
      version = "~> 0.0.1"
    }
  }
}

# Configure the provider
provider "latitudesh" {
  auth_token = <YOUR_LATITUDESH_TOKEN>
}
```
and run `terraform init`. Now you are ready to start using terraform.
